Add spec

marks1024 opened the pull request:
Since the plan is to refactor the grammar (as discussed in #97) I thought it would be useful to have a spec/test suite.  So I added one.  This is a minimal working version but should be easily extensible.  Spec passes on my machine.

yudai-nkt commented on CHANGELOG.md:
```diff
@@ -6,6 +6,7 @@
 - Several file extensions are interpreted as LaTeX file ([#81](https://github.com/area/language-latex/pull/81), [#82](https://github.com/area/language-latex/pull/82), and
 [#126](https://github.com/area/language-latex/pull/126))
 - Support for highlighting long lines ([#131](https://github.com/area/language-latex/pull/131))
+- Added spec
```
Move this line on top of the file. Your PR will be published in a future release, not in v1.1.0, which is already out. Also please use imperative mood instead of past tense.

yudai-nkt commented on spec/latex-test-spec.coffee:
```diff
@@ -0,0 +1,25 @@
+      tk = grammar.tokenizeLines(s)
+      expect(tk[0][14].scopes[1]).toBe 'string.other.math.tex'
+
+   it "gets the language correct (python)", ->
```
As I'm not so familiar with how to write a spec, I'd like to keep it simple as a starting point at first. Would you please remove this listing section if you don't mind?

yudai-nkt commented:
I really appreciate this because I wanted a spec but don't know how to start as in yudai-nkt/language-tex#7. I have some comments outside the review.

1. I'm aware that test suites are powerful tools for software development, but I'm afraid I can't relate grammar refactoring (or #97) to the usefulness of a spec. Could you explain?

1. After some investigation on [Atom's official language packages](https://github.com/atom?&q=language-), `<language-name>-spec.coffee` looks like a popular naming convention. So, could you rename the spec to `latex-spec.coffee`? (this should be a review, but it seems we can't review on file names.)

marks1024 commented:
I made a new commit that I believe addresses all issues.  

Regarding point 1.  I was looking into the `repository` feature and it will require some rearrangement to the grammar file, which could easily introduce some bugs.  I thought having a spec would give collaborators some confidence that they aren't breaking anything during the refactor.  And it will be quicker than checking manually on sample latex files.  Although, I'm not sure we can test everything this way, for example snippets.

yudai-nkt commented:
Please resolve conflicts.

yudai-nkt commented:
Thanks. will take care of indentation on my end.
